API: output for [3rd-party] check-update

castulo opened the pull request:
This commit provides a minimal output to be displayed when the --quiet
flag is used for these commands:
- swupd check-update
- swupd 3rd-party check-update

Signed-off-by: Castulo Martinez <castulo.martinez@intel.com>

otaviobp commented on src/3rd_party_repos.c:
```diff
@@ -543,9 +543,13 @@ enum swupd_code third_party_run_operation_multirepo(const char *repo, process_bu
 void third_party_repo_header(const char *repo_name)
 {
 	char *header = NULL;
+	bool quiet = (log_get_level() == LOG_ERROR);
```
This is hard to maintain.
I would create a `#define LOG_QUIET LOG_ERROR` in log.h or some kind of function log_is_debug() and log_is_quiet().
And the ideal would be `log_get_level() < LOG_ERROR` to include LOG_ALWAYS too.

castulo replied:
good idea

otaviobp commented on src/check_update.c:
```diff
@@ -120,11 +120,12 @@ enum swupd_code check_update(void)
 
 	if (current_version < server_version) {
-		print("There is a new OS version available: %d\n", server_version);
+		info("There is a new OS version available: ", server_version);
```
extra parameter

castulo replied:
oops!

otaviobp commented on test/functional/api/api-3rd-party-checkupdate.bats:
```diff
@@ -0,0 +1,57 @@
+	expected_output=$(cat <<-EOM
+		[repo1]
+		20
+		[repo2]
```
I am unsure about this. Should we print the header [repo2] if there's no update?

castulo replied:
It would look nicer if we didn't, but I think is a limitation we have. I think it would be very difficult to not print it, I couldn't think of a way of doing it without doing a massive refactoring.

castulo commented:
V2: addressed all the comments except the one regarding removing the [repo_name] from the repo that did not have output. As I said, I think this is a limitation, it would look better if we didn't print the repos that don't have output, but I don't think it can be done without having to do a big refactoring, and IMO is not worth it.
